Lib/importlib/_bootstrap_external.py

```diff
@@ -1711,6 +1711,78 @@ def __repr__(self):
         return f'FileFinder({self.path!r})'
 
 
+class AppleFrameworkLoader(ExtensionFileLoader):
+    """A loader for modules that have been packaged as frameworks for
+    compatibility with Apple's iOS App Store policies.
+
+    For compatibility with the iOS App Store, *all* binary modules in an iOS
+    app must be .dylib objects, contained in a framework with appropriate
+    metadata, stored in the ``Frameworks`` folder of the packaged app. There
+    can be only a single binary per framework, and there can be no executable
+    binary material outside the Frameworks folder.
+
+    If you're trying to run ``from foo.bar import _whiz``, and ``_whiz`` is
+    implemented with the binary module ``foo/bar/_whiz.abi3.dylib`` (or any
+    other ABI .dylib extension), this loader will look for
+    ``{sys.executable}/Frameworks/foo.bar._whiz.framework/_whiz.abi3.dylib``
+    (forming the package name by taking the full import path of the library,
+    and replacing ``/`` with ``.``).
+
+    However, the ``__file__`` attribute of the ``_whiz`` module will report as
+    the original location inside the ``foo/bar`` subdirectory. This so that
+    code that depends on walking directory trees will continue to work as
+    expected based on the *original* file location.
+
+    The Xcode project building the app is responsible for converting any
+    ``.dylib`` files from wherever they exist in the ``PYTHONPATH`` into
+    frameworks in the ``Frameworks`` folder (including the addition of
+    framework metadata, and signing the resulting framework). This will usually
+    be done with a build step in the Xcode project; see the iOS documentation
+    for details on how to construct this build step.
+    """
```

Comment:
This appears to be mostly a copy-and-paste of what you've added to the importlib docs.  I'd say keep the docstring here brief.  We can rely on the docs for all these details.

```suggestion
    """A loader for modules that have been packaged as frameworks for
    compatibility with Apple's iOS App Store policies.
    """
```

If there are any implementation notes that don't belong in the docs then they would make more sense as comments here than as part of the docstring.  However, I don't see any such notes.

If you do strip this down, please make sure to update the docs with any differences between the two, as appropriate.  I noticed that there are some differences.

Reply:
That makes sense. There's very little in these docs that is specific to the actual implementation; it's all high level "this is why iOS is weird" stuff.

Any differences that exist are purely because I did some editing in one (or both) versions, and neglected to reflect those edits in the other version. It's a moot point once they've been removed here, but I'll make sure I don't lose anything useful from the deleted version.